app/scripts/lib/rpc-method-middleware/handlers/types.ts

```diff
@@ -0,0 +1,4 @@
+export type HandlerWrapper = {
+  methodNames: [string] | string[];
```

Comment:
Would it make sense to narrow this to some range of valid method names?

```suggestion
  methodNames: [MESSAGE_TYPE] | MESSAGE_TYPE[];
```

Reply:
I tired to do as per suggestion but MESSAGE_TYPE is not getting accepted as a valid type.

Reply:
Could we try this?

```suggestion
  methodNames:
    | [(typeof MESSAGE_TYPE)[keyof typeof MESSAGE_TYPE]]
    | (typeof MESSAGE_TYPE)[keyof typeof MESSAGE_TYPE][];
```

Feel free to add any other types that might fit here.

Reply:
This might be more readable:

```diff
diff --git a/app/scripts/lib/rpc-method-middleware/handlers/types.ts b/app/scripts/lib/rpc-method-middleware/handlers/types.ts
index f2ae5a3db5..5b7a2a7494 100644
--- a/app/scripts/lib/rpc-method-middleware/handlers/types.ts
+++ b/app/scripts/lib/rpc-method-middleware/handlers/types.ts
@@ -1,4 +1,6 @@
+import { MessageType } from '../../../../../shared/constants/app';
+
 export type HandlerWrapper = {
-  methodNames: [string] | string[];
+  methodNames: [MessageType] | MessageType[];
   hookNames: Record<string, boolean>;
 };
diff --git a/shared/constants/app.ts b/shared/constants/app.ts
index a38ac274d3..b0d9ebd158 100644
--- a/shared/constants/app.ts
+++ b/shared/constants/app.ts
@@ -65,6 +65,8 @@ export const MESSAGE_TYPE = {
   ///: END:ONLY_INCLUDE_IF
 } as const;
 
+export type MessageType = (typeof MESSAGE_TYPE)[keyof typeof MESSAGE_TYPE];
+
 export const SNAP_DIALOG_TYPES = {
   [DialogType.Alert]: MESSAGE_TYPE.SNAP_DIALOG_ALERT,
   [DialogType.Confirmation]: MESSAGE_TYPE.SNAP_DIALOG_CONFIRMATION,

```